provider/virt_storage/storage_volume.py

```diff
@@ -75,6 +75,9 @@ def format(self, format):
         else:
             format_cls = qdevices.QBlockdevFormatRaw
         self._format = format_cls(self.name)
+        if (self._protocol is not None and
```

Comment:
hi @luckyh , I fixed this issue by adding line160-162, the patch was merged yesterday, do we still have issues?

Reply:
https://github.com/autotest/tp-qemu/pull/4118

Reply:
@zhencliu sorry I didn't note about #4118. However, it seems that we'd better handle the node parentship right after object instantiation, otherwise, there will be somewhere could trigger this issue too.

What do you think?

Reply:
Never mind : )
These methods are private, which are called by another private method refresh_with_params, then the private refresh_with_params is called by the public method volume_define_by_params , which is the only entrance when defining a volume, it could be safe.
However, it's OK for me to add these in case one day some one may call the functions in a bad way